src/snaplet.ts

```diff
@@ -0,0 +1,608 @@
+        name: columns.shift(),
+        status: columns.shift().includes("SUCCESS") ? "SUCCESS" : "ERROR",
+        created: `${columns.shift()} ${columns.shift()} ${columns.shift()}`,
+        size: `${columns.shift()}${columns.shift()}`,
+        src: columns.shift().includes("☁️") ? "CLOUD" : "LOCAL",
+      });
+    }
+  }
+  return result;
+}
+
+function parsePreviewDatabaseList(output: string): PreviewDatabase[] {
+  const result: PreviewDatabase[] = [];
+  const outputSplit = output
+    // Trim all multiples spaces into only single one
+    .replace(/ +/g, " ")
+    // Remove all terminal escaping sequence
+    .replace(/\u001b\[[0-9;]*[A-Za-z]/g, "")
+    .split("\n");
+  // if there is at least one database
+  if (outputSplit.length > 2 && outputSplit[1].startsWith("NAME")) {
+    // Skip the header lines
+    for (let i = 2; i < outputSplit.length; i++) {
+      // Each database have a name, size and the snapshot used for creation
+      const columns = outputSplit[i].split(" ");
+      const name = columns.shift();
+      const size = `${columns.shift()}${columns.shift()}`;
+      const snapshot = columns.shift();
+      result.push({
+        name,
+        size,
+        snapshotName: snapshot,
+      });
+    }
+  }
+  return result;
+}
+
+const snapshotsGenerator: Fig.Generator = {
+  script: "snaplet snapshot ls",
+  postProcess: (output) => {
+    const result: Fig.Suggestion[] = [];
+    const snapshotList = parseSnapshotList(output);
+    for (const snapshot of snapshotList) {
+      const statusEmoji = snapshot.status === "SUCCESS" ? `✅` : `❌`;
+      const locationEmoji = snapshot.src === "CLOUD" ? "☁️" : "💻";
+      result.push({
+        name: snapshot.name,
+        description: `${statusEmoji} ${snapshot.created} ${snapshot.size} ${locationEmoji}`,
+      });
+    }
+    return result;
+  },
+};
+
+// Only suggest successful snapshots on the cloud
+const snapshotsSuccessCloudGenerator: Fig.Generator = {
+  script: "snaplet snapshot ls",
+  postProcess: (output) => {
+    const result: Fig.Suggestion[] = [];
+    const snapshotList = parseSnapshotList(output).filter(
+      (s) => s.src === "CLOUD" && s.status === "SUCCESS"
+    );
+    for (const snapshot of snapshotList) {
+      const statusEmoji = `✅`;
+      const locationEmoji = "☁️";
+      result.push({
+        name: snapshot.name,
+        description: `${statusEmoji} ${snapshot.created} ${snapshot.size} ${locationEmoji}`,
+      });
+    }
+    return result;
+  },
+};
+
+const databaseGenerator: Fig.Generator = {
+  script: "snaplet database ls",
+  postProcess: (output) => {
+    const result: Fig.Suggestion[] = [];
+    const databases = parsePreviewDatabaseList(output);
+    for (const database of databases) {
+      result.push({
+        name: database.name,
+        description: `${database.size} (${database.snapshotName})`,
+      });
+    }
+    return result;
+  },
+};
+
+const commonCommandsOptions = [
+  {
+    name: ["--help", "-h"],
```

Comment:
With this property the option becomes automatically available to all subcommands. In such way you don't have to worry about adding `options: [...commonCommandsOptions]`. Can you please remove those?
```suggestion
    name: ["--help", "-h"],
    isPersistent: true,
```

Reply:
Oh nice ! I was sure they were a better way to do it. Thank's for the suggestion !